docs/feature-gates.md

```diff
@@ -188,3 +189,53 @@ Node with IP Address 10.10.10.10.
 #### Requirements for this Feature
 
 This feature is currently only supported for Nodes running Linux with IPv4 addresses.
+
+### Egress
+
+`Egress` enables a CRD API for Antrea that supports specifying which egress
+(SNAT) IP the traffic from the selected Pods to the external network should use.
+When a selected Pod accesses the external network, the egress traffic will be
+tunneled to the Node that hosts the egress IP if it's different from the Node
+that the Pod runs on and will be SNATed to the egress IP when leaving that Node.
+Usage example:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha2
+kind: Egress
+metadata:
+  name: egress-web
+spec:
+  appliedTo:
+    podSelector:
+      matchLabels:
+        role: web
+    namespaceSelector:
+      matchLabels:
+        env: prod
+  egressIP: 10.0.10.8
+```
+
+The `appliedTo` field specifies the grouping criteria of Pods to which the
+Egress applies to. Pods can be selected cluster-wide using `podSelector`. If set
+with a `namespaceSelector`, Pods from Namespaces selected by the
+namespaceSelector will be selected. Empty `appliedTo` selects nothing. The field
+is mandatory.
+
+The `egressIP` field specifies the egress (SNAT) IP the traffic from the
+selected Pods to the external network should use. **The IP must be assigned to
+an arbitrary interface of one Node, and one Node only. It must be reachable from
+all Nodes.** For IPv4 cluster, it must be an IPv4 address; for IPv6 cluster, it
+must be an IPv6 address. The field is mandatory.
+
+**Note**: If more than one Egress applies to a Pod and they specify different
+`egressIP`, the effective egress IP will be selected randomly.
+
+In the above example, the Egress applies to Pods which match the labels
+"role=web" from Namespaces which match the labels "env=prod". The source IPs of
+their egress traffic to external network will be translated to 10.0.10.8.
+
+#### Requirements for this Feature
+
+This feature is currently only supported for Nodes running Linux and "encap"
+mode. The support for Windows and other traffic modes will be added in the
```

Comment:
So you believe it does not work for hybrid now? I thought it should work.

Reply:
I think it again and agree it should work. I could test it later and update the doc after verifying it.

Reply:
Sounds good!

Reply:
@jianjuns It didn't work for two Nodes in same subnet in hybrid case. The reason is:
1. After the request packet is tunneled to the Egress Node and sent to its host network, reverse path filtering will drop the packet as the traffic comes from antrea-gw0 while it's on a route of one physical interface like eth0. rp_filter is set to strict mode in some distributions like ubuntu. Changing it to loose mode can resolve this.
2. After the reply packet is received on the Egress Node, it will output the packet to the physical interface that has route to the Pod network, e.g. eth0, instead of antrea-gw0, leading to asymmetric path.

I will file an issue to discuss how to make it work for hybrid mode, and maybe route mode together. For this release, guess we don't have time to support it for hybrid mode. Do you agree?

Reply:
Got it. Let us think about any solution. I think the feature can be useful for hybrid mode and encap mode too.

Reply:
sure